### ISSUE TITLE
vue/component-api-style probably should not report "extends"

### ISSUE THREAD

**Issue by ilyub** [vuejs/eslint-plugin-vue#1801]
**What rule do you want to change?**

vue/component-api-style

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

It should not report "extends" for composition api because there is no alternative.

**Please provide some example code that this change will affect:**

```vue
eslintrc.js
"vue/component-api-style": ["warn", ["composition"]],

comp1.vue:
export default defineComponent({
  name: "comp-1",
  extends: Comp2,
```

**What does the rule currently do for this code?**

Shows warning for "extends" section. 

**What will the rule do after it's changed?**

It should not report "extends" for composition API because there is no alternative.

**Comment by ota-meshi** [comment]
Thank you for posting this issue.

As far as I know, you can't use `extends` with `__VUE_OPTIONS_API__: false`. So if you only want to use the Composition API, I don't think you should use `extends`.
https://github.com/vuejs/core/blob/main/packages/vue/README.md#bundler-build-feature-flags
So I think the rule is working correctly.
(Please let me know if the flag specifications I know are incorrect.)

If you want to mix the Composition API and the Options API, you may need to change the options in that rule to allow the Options API.
